packages/sdk/src/base/auth.ts

```diff
@@ -84,45 +90,45 @@ export class Auth implements IAuth {
 		this.storage.auth_token = result.access_token;
 		this.storage.auth_refresh_token = result.refresh_token ?? null;
 		if (result.expires) {
-			this.storage.auth_expires = Date.now() + result.expires;
+			// should technically be the response time of the request, not Date.now()
+			this.storage.auth_expires_at = Date.now() + result.expires;
 		} else {
-			this.storage.auth_expires = null;
+			this.storage.auth_expires_at = null;
 		}
 	}
 
-	private updateRefresh(options?: Partial<AuthRefreshOptions>) {
-		const expiration = this.storage.auth_expires;
-		if (expiration === null) {
-			clearTimeout(this.timer as ReturnType<typeof setTimeout>);
-			return; // Don't auto refresh if there's no expiration time (token auth)
+	private updateRefresher(options?: Partial<AuthAutoRefreshOptions>) {
+		// Reset timer
+		clearTimeout(this.timer as ReturnType<typeof setTimeout>);
+
+		const expiresAt = this.storage.auth_expires_at;
+		if (expiresAt === null) {
+			return; // Don't set timer if there's no expiration time (token auth)
 		}
 
+		// Update options
 		if (options) {
-			this.options.refresh!.auto = options.auto ?? this.options.refresh!.auto;
-			this.options.refresh!.time = options.time ?? this.options.refresh!.time;
+			this.options.refreshOptions!.autoRefresh = options.autoRefresh ?? this.options.refreshOptions!.autoRefresh;
+			this.options.refreshOptions!.autoRefreshLeadTime =
+				options.autoRefreshLeadTime ?? this.options.refreshOptions!.autoRefreshLeadTime;
 		}
 
-		clearTimeout(this.timer as ReturnType<typeof setTimeout>);
-
-		let remaining = expiration - this.options.refresh!.time! - Date.now();
+		let remaining = expiresAt - this.options.refreshOptions!.autoRefreshLeadTime! - Date.now();
+		// TODO: how to deal with the unlikely edge case that LeadTime > token TTL; creates infinite refresh loop (why doesn't debouncer catch this?)
 		if (remaining < 0) {
 			// It's already expired, try a refresh
-			if (expiration < Date.now()) {
-				return; // Don't set auto refresh
+			if (expiresAt < Date.now()) {
+				// Is already expired
+				return;
```

Comment:
I'm not entirely sure why this returns immediately. It didn't seem to make a difference taking it out, but I could be missing something.